src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -91,6 +91,45 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException(SR.Arg_BasePathNotFullyQualified);
+
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
```

Comment:
>if [](start = 12, length = 2)

You need to check the inputs for embedded nulls at this point and throw as above. We don't ever want to give back "c:\\Foo\\\0bar".